integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/Utils.java

```diff
@@ -410,10 +410,18 @@ public static boolean isCarFileDeployed(String carFileName, ApplicationAdminClie
     public static void deploySynapseConfiguration(OMElement config, String artifactName, String artifactType,
                                                   boolean isRestartRequired) {
 
-        String path = System.getProperty("carbon.home") + File.separator + "repository" + File.separator + "deployment"
+        String directory = System.getProperty("carbon.home") + File.separator + "repository" + File.separator + "deployment"
                 + File.separator + "server" + File.separator + "synapse-configs" + File.separator + "default"
-                + File.separator + artifactType + File.separator + artifactName + ".xml";
+                + File.separator + artifactType;
+        String path = directory + File.separator + artifactName + ".xml";
 
+        if (!Files.exists(FileSystems.getDefault().getPath(directory))) {
+            try {
+                Files.createDirectories(FileSystems.getDefault().getPath(directory));
+            } catch (IOException e) {
+                log.error("Error while creating the directory, " + directory + ".", e);
```

Comment:
Shouldn't we throw the exception instead of catching and logging since it does not make sense to proceed without creating the directory?

Reply:
Fixed in https://github.com/wso2/micro-integrator/pull/580